rotkehlchen/chain/evm/manager.py

```diff
@@ -0,0 +1,1008 @@
+                        log.warning(msg)
+                        synchronized = False
+                    else:
+                        synchronized, msg = _is_synchronized(current_block, latest_block)
+            except ValueError as e:
+                message = (
+                    f'Failed to connect to {self.chain_name} node {node} at endpoint '
+                    f'{rpc_endpoint} due to {str(e)}'
+                )
+                return False, message
+
+            if not synchronized:
+                self.msg_aggregator.add_warning(
+                    f'We could not verify that {self.chain_name} node {node} is '
+                    'synchronized with mainnet. Balances and other queries '
+                    'may be incorrect.',
+                )
+
+            log.info(f'Connected {self.chain_name} node {node} at {rpc_endpoint}')
+            self.web3_mapping[node] = web3
+            return True, ''
+
+        # else
+        message = f'Failed to connect to {self.chain_name} node {node} at endpoint {rpc_endpoint}'
+        log.warning(message)
+        return False, message
+
+    def connect_to_multiple_nodes(self, nodes: Sequence[WeightedNode]) -> None:
+        self.web3_mapping = {}
+        for weighted_node in nodes:
+            if weighted_node.node_info.name == self.etherscan_node_name:
+                continue
+
+            task_name = f'Attempt connection to {str(weighted_node.node_info.name)} {self.chain_name} node'  # noqa: E501
+            self.greenlet_manager.spawn_and_track(
+                after_seconds=None,
+                task_name=task_name,
+                exception_is_error=True,
+                method=self.attempt_connect,
+                node=weighted_node.node_info,
+                mainnet_check=True,
+            )
+
+    def query(self, method: Callable, call_order: Sequence[WeightedNode], **kwargs: Any) -> Any:
+        """Queries evm related data by performing the provided method to all given nodes
+
+        The first node in the call order that gets a successful response returns.
+        If none get a result then RemoteError is raised
+        """
+        for weighted_node in call_order:
+            node = weighted_node.node_info
+            web3 = self.web3_mapping.get(node, None)
+            if web3 is None and node.name != self.etherscan_node_name:
+                continue
+
+            try:
+                result = method(web3, **kwargs)
+            except (
+                    RemoteError,
+                    requests.exceptions.RequestException,
+                    BlockchainQueryError,
+                    TransactionNotFound,
+                    BlockNotFound,
+                    BadResponseFormat,
+                    ValueError,  # Yabir saw this happen with mew node for unavailable method at node. Since it's generic we should replace if web3 implements https://github.com/ethereum/web3.py/issues/2448  # noqa: E501
+            ) as e:
```

Comment:
isn't this over indented?

Reply:
Oops it was like this before! I have fixed it.